rad/src/error.rs

```diff
@@ -117,6 +117,12 @@ pub enum RadError {
     /// Overflow error
     #[fail(display = "Overflow error")]
     Overflow,
+    /// Mismatching types
+    #[fail(display = "Mismatching types")]
```

Comment:
This error message is lacking context. We should print both values for debugging purposes.